Tests/SonarScanner.MSBuild.PreProcessor.Test/ArchiveDownloaderTests.cs

```diff
@@ -835,35 +830,30 @@ public async Task EndToEndTestWithFiles_TarGz_Success()
                 Path.Combine(cache, sha, $"{file}_extracted", "jdk-17.0.11+9-jre", "bin", "java.exe"));
             File.ReadAllText(Path.Combine(cache, sha, $"{file}_extracted", "jdk-17.0.11+9-jre", "bin", "java.exe")).Should().Be(
                 "This is just a sample file for testing and not the real java.exe");
-            runtimeIO.Logger.Should().HaveInfoOnce("""
-                The JRE provisioning is a time consuming operation.
-                JRE provisioned: OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz.
-                If you already have a compatible Java version installed, please add either the parameter "/d:sonar.scanner.skipJreProvisioning=true" or "/d:sonar.scanner.javaExePath=<PATH>".
-                """);
             runtimeIO.Logger.DebugMessages.Should().SatisfyRespectively(
                 x => x.Should().Be($"Cache miss. Attempting to download '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz")}'."),
                 x => x.Should().Be($"The checksum of the downloaded file is '{sha}' and the expected checksum is '{sha}'."),
-                x => x.Should().Match($"Starting extracting the Java runtime environment from archive '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz")}' to folder '{Path.Combine(cache, sha, "*")}'."),
-                x => x.Should().Match($"Moving extracted Java runtime environment from '{Path.Combine(cache, sha, "*")}' to '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz_extracted")}'."),
-                x => x.Should().Be($"The Java runtime environment was successfully added to '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz_extracted")}'."));
+                x => x.Should().Match($"Starting to extract files from archive '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz")}' to folder '{Path.Combine(cache, sha, "*")}'."),
+                x => x.Should().Match($"Moving extracted files from '{Path.Combine(cache, sha, "*")}' to '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz_extracted")}'."),
+                x => x.Should().Be($"The archive was successfully extracted to '{Path.Combine(cache, sha, "OpenJDK17U-jre_x64_windows_hotspot_17.0.11_9.tar.gz_extracted")}'."));
         }
         finally
         {
             Directory.Delete(home, true);
         }
     }
 
-    private async Task<DownloadResult> ExecuteDownloadAndUnpack(MemoryStream content = null, JreDescriptor descriptor = null)
+    private async Task<DownloadResult> ExecuteDownloadAndUnpack(MemoryStream content = null, ArchiveDescriptor descriptor = null)
     {
-        var jreDescriptor = descriptor ?? new JreDescriptor("filename.tar.gz", "sha256", "javaPath");
-        var sut = CreateSutWithSubstitutes(jreDescriptor);
+        var archiveDescriptor = descriptor ?? new ArchiveDescriptor("filename.tar.gz", "sha256", "target file");
```

Comment:
Can we give this a value that resembles a filename, like e.g. `InsideArchive.exe` or  `InsideArchive.jar` or `InsideArchive.dll`
```suggestion
        var archiveDescriptor = descriptor ?? new ArchiveDescriptor("filename.tar.gz", "sha256", "InsideArchive.exe");
```